src/net_processing.cpp

```diff
@@ -889,15 +889,24 @@ void PeerManager::InitializeNode(CNode *pnode) {
 
 void PeerManager::ReattemptInitialBroadcast(CScheduler& scheduler) const
 {
-    std::map<uint256, uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
-
-    for (const auto& elem : unbroadcast_txids) {
-        // Sanity check: all unbroadcast txns should exist in the mempool
-        if (m_mempool.exists(elem.first)) {
-            LOCK(cs_main);
+    std::vector<std::pair<uint256, uint256>> relay_transactions;
+    {
+        LOCK(m_mempool.cs);
+        std::map<uint256, uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
```

Comment:
This approach is also a small step towards removing the lock from `GetUnbroadcastTxs` and `exists`.

Reply:
Done in fb2dca415f6ac74d79292296831ccc33c485230b and dde441c11a07b9ff72dbcfda896c61adf33ae228.